workspace/utilities/lib/master-title.xsl

```diff
@@ -1,55 +1,75 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+<xsl:stylesheet version="1.0" 
+	xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+	xmlns:exslt="http://exslt.org/common"
+	exclude-result-prefixes="exslt">
 
-<!-- Template to override for custom title -->
+<!-- Template to override for custom master title algo -->
 	<xsl:template name="master-title">
-		<xsl:call-template name="default-master-title" />
+		<xsl:param name="lg" select="$url-language" />
+
+		<xsl:call-template name="default-master-title" >
+			<xsl:with-param name="lg" select="$lg" />
+		</xsl:call-template>
+	</xsl:template>
+
+<!-- Override in the page to stop appending site name to the page-title -->
+	<xsl:variable name="master-title-append-site-name" select="string-length($page-metas/titre) = 0" />
+
+<!-- Default master title template with default algo for page title and site name -->
+	<xsl:template name="default-master-title">
+		<xsl:param name="lg" select="$url-language" />
+
+		<xsl:if test="count(/data/params/page-types/item[@handle = 'index']) = 0">
```

Comment:
We should not have a special case for the index. It is not needed anymore.

Reply:
We still need a special case when no custom meta are set

Reply:
Yes use the default in site-config. But let the index have a different meta then the site-config.